src/plugins/intel_npu/src/backend/include/zero_variable_state.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "intel_npu/common/variable_state.hpp"
+#include "intel_npu/config/config.hpp"
+#include "intel_npu/network_metadata.hpp"
+#include "intel_npu/utils/logger/logger.hpp"
+#include "intel_npu/utils/zero/zero_init.hpp"
+
+namespace intel_npu {
+
+class ZeroVariableState final : public VariableState {
```

Comment:
```suggestion
class ZeroVariableState final : public ov::IVariableState {
```
Any reason not to inherit `ov::IVariableState` directly instead of this one, since you're borrowing nothing from `intel_npu::VariableState` anyway? Any constraints? No hard opinion, just make sure the class hierarchy makes sense.

Reply:
Documentation also. Why a new class, how is it different from `VariableState` etc. Docs for all custom methods too.

Reply:
Yeah, that would make sense

Reply:
done, thx